tasks_management/schema.py

```diff
@@ -0,0 +1,72 @@
+import graphene
+
+from django.contrib.auth.models import AnonymousUser
+from django.db.models import Q
+
+from core.schema import OrderedDjangoFilterConnectionField
+from core.utils import append_validity_filter
+from tasks_management.apps import TasksManagementConfig
+from tasks_management.gql_mutations import CreateTaskGroupMutation
+from tasks_management.gql_queries import TaskGroupGQLType, TaskExecutorGQLType
+from tasks_management.models import TaskGroup, TaskExecutor
+
+import graphene_django_optimizer as gql_optimizer
+
+
+class Query(graphene.ObjectType):
+    module_name = "tasks_management"
+
+    task_group = OrderedDjangoFilterConnectionField(
+        TaskGroupGQLType,
+        orderBy=graphene.List(of_type=graphene.String),
+        dateValidFrom__Gte=graphene.DateTime(),
+        dateValidTo__Lte=graphene.DateTime(),
+        client_mutation_id=graphene.String(),
+    )
+    task_executor = OrderedDjangoFilterConnectionField(
+        TaskExecutorGQLType,
+        orderBy=graphene.List(of_type=graphene.String),
+        dateValidFrom__Gte=graphene.DateTime(),
+        dateValidTo__Lte=graphene.DateTime(),
+        client_mutation_id=graphene.String(),
+    )
+
+    def resolve_task_group(self, info, **kwargs):
+        filters = append_validity_filter(**kwargs)
+
+        client_mutation_id = kwargs.get("client_mutation_id", None)
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        Query._check_permissions(
+            info.context.user,
+            TasksManagementConfig.gql_task_group_search_perms
+        )
+        query = TaskGroup.objects.filter(*filters)
+        return gql_optimizer.query(query, info)
+
+    def resolve_task_executor(self, info, **kwargs):
+        filters = append_validity_filter(**kwargs)
+
+        client_mutation_id = kwargs.get("client_mutation_id")
+        if client_mutation_id:
+            filters.append(Q(mutations__mutation__client_mutation_id=client_mutation_id))
+
+        task_group_id = kwargs.get("taskGroupId")
+        if task_group_id:
+            filters.append(Q(taskgroup__user__id=task_group_id))
```

Comment:
Shouldn't this be resolved by the GraphQL model definition? 
This piece of code 
```python
**prefix_filterset("task_group__", TaskGroupGQLType._meta.filter_fields),
```
should create binding. 
Additionally I think that fields that are not part of the "core" definition coming from the model have to be listed in GQL Field: 
```python
    task_executor = OrderedDjangoFilterConnectionField(
        TaskExecutorGQLType,
        orderBy=graphene.List(of_type=graphene.String),
        dateValidFrom__Gte=graphene.DateTime(),
        dateValidTo__Lte=graphene.DateTime(),
        client_mutation_id=graphene.String(),
    )
```
Otherwise I'm not sure if they'll be included in the generated documentation.

Reply:
You are right. I am removing this part of code. I didn't add the definition because I saw it in documentation and it got me tricked.